### ISSUE TITLE
Crash when linting kitchen-async

### ISSUE THREAD

**Issue by lucywang000** [clj-kondo/clj-kondo#1148]
**version**

```
$ clj-kondo --version
clj-kondo v2021.01.20
```

**platform**

MacOS 10.15

**problem repro**

```sh
$ git clone git@github.com:athos/kitchen-async.git
$ cd kitchen-async
$ clj-kondo --lint src/kitchen_async/promise.cljc
Unexpected error. Please report an issue.
java.lang.NullPointerException
        at clojure.lang.Numbers.ops(Numbers.java:1068)
        at clojure.lang.Numbers.gt(Numbers.java:259)
        at clj_kondo.impl.linters$lint_var_usage.invokeStatic(linters.clj:278)
        at clj_kondo.core$run_BANG_.invokeStatic(core.clj:136)
        at clj_kondo.main$main.invokeStatic(main.clj:117)
        at clj_kondo.main$main.doInvoke(main.clj:113)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:665)
        at clj_kondo.main$_main$fn__15684.invoke(main.clj:136)
        at clj_kondo.main$_main.invokeStatic(main.clj:136)
        at clj_kondo.main$_main.doInvoke(main.clj:134)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clj_kondo.main.main(Unknown Source)
```

This is the cljc file being linted: https://github.com/athos/kitchen-async/blob/1f28eab/src/kitchen_async/promise.cljc

Btw, I hit this bug when using clojure-lsp, here is the stack trace for that

<details>

```clojure
2021/01/27 21:11:25 29237 INFO  clojure-lsp.crawler Analyzing 337 paths with clj-kondo with batch size of  7 ...
2021/01/27 21:11:25 29237 INFO  clojure-lsp.crawler Analyzing 1/7 batch paths with clj-kondo...
2021/01/27 21:11:32 29237 ERROR clojure-lsp.main #error {
 :cause nil
 :via
 [{:type java.lang.NullPointerException
   :message nil
   :at [clojure.lang.Numbers ops Numbers.java 1068]}]
 :trace
 [[clojure.lang.Numbers ops Numbers.java 1068]
  [clojure.lang.Numbers gt Numbers.java 259]
  [clj_kondo.impl.linters$lint_var_usage invokeStatic linters.clj 278]
  [clj_kondo.impl.linters$lint_var_usage invoke linters.clj 205]
  [clj_kondo.core$run_BANG_ invokeStatic core.clj 143]
  [clj_kondo.core$run_BANG_ invoke core.clj 51]
  [clojure_lsp.crawler$run_kondo_on_paths_BANG_ invokeStatic crawler.clj 93]
  [clojure_lsp.crawler$run_kondo_on_paths_BANG_ invoke crawler.clj 92]
  [clojure_lsp.crawler$run_kondo_on_paths_batch_BANG_$fn__27849 invoke crawler.clj 110]
  [clojure.core$map_indexed$mapi__8548$fn__8549 invoke core.clj 7310]
  [clojure.lang.LazySeq sval LazySeq.java 42]
  [clojure.lang.LazySeq seq LazySeq.java 51]
  [clojure.lang.RT seq RT.java 535]
  [clojure.core$seq__5402 invokeStatic core.clj 137]
  [clojure.core.protocols$seq_reduce invokeStatic protocols.clj 24]
  [clojure.core.protocols$fn__8146 invokeStatic protocols.clj 75]
  [clojure.core.protocols$fn__8146 invoke protocols.clj 75]
  [clojure.core.protocols$fn__8088$G__8083__8101 invoke protocols.clj 13]
  [clojure.core$reduce invokeStatic core.clj 6828]
  [clojure.core$into invokeStatic core.clj 6895]
  [clojure.core$into invoke core.clj 6887]
  [clojure_lsp.crawler$run_kondo_on_paths_batch_BANG_ invokeStatic crawler.clj 111]
  [clojure_lsp.crawler$run_kondo_on_paths_batch_BANG_ invoke crawler.clj 96]
  [clojure_lsp.crawler$analyze_paths invokeStatic crawler.clj 154]
  [clojure_lsp.crawler$analyze_paths invoke crawler.clj 151]
  [clojure_lsp.crawler$analyze_classpath invokeStatic crawler.clj 183]
  [clojure_lsp.crawler$analyze_classpath invoke crawler.clj 167]
  [clojure_lsp.crawler$analyze_project invokeStatic crawler.clj 190]
  [clojure_lsp.crawler$analyze_project invoke crawler.clj 186]
  [clojure_lsp.crawler$initialize_project invokeStatic crawler.clj 226]
  [clojure_lsp.crawler$initialize_project invoke crawler.clj 215]
  [clojure_lsp.handlers$initialize invokeStatic handlers.clj 105]
  [clojure_lsp.handlers$initialize invoke handlers.clj 103]
  [clojure.lang.Var invoke Var.java 393]
  [clojure_lsp.main$fn__30484$fn__30485 invoke main.clj 308]
  [clojure_lsp.main.proxy$clojure_lsp.ClojureExtensions$LanguageServer$c8d5825a initialize nil -1]
  [jdk.internal.reflect.NativeMethodAccessorImpl invoke0 NativeMethodAccessorImpl.java -2]
  [jdk.internal.reflect.NativeMethodAccessorImpl invoke NativeMethodAccessorImpl.java 62]
  [jdk.internal.reflect.DelegatingMethodAccessorImpl invoke DelegatingMethodAccessorImpl.java 43]
  [java.lang.reflect.Method invoke Method.java 566]
  [org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint lambda$null$0 GenericEndpoint.java 65]
  [org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint request GenericEndpoint.java 120]
  [org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleRequest RemoteEndpoint.java 261]
  [org.eclipse.lsp4j.jsonrpc.RemoteEndpoint consume RemoteEndpoint.java 190]
  [org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer handleMessage StreamMessageProducer.java 194]
  [org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer listen StreamMessageProducer.java 94]
  [org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor run ConcurrentMessageProcessor.java 113]
  [java.util.concurrent.Executors$RunnableAdapter call Executors.java 515]
  [java.util.concurrent.FutureTask run FutureTask.java 264]
  [java.util.concurrent.ThreadPoolExecutor runWorker ThreadPoolExecutor.java 1128]
  [java.util.concurrent.ThreadPoolExecutor$Worker run ThreadPoolExecutor.java 628]
  [java.lang.Thread run Thread.java 834]]}
```

</details>

**Comment by borkdude** [comment]
Fixed with ebbf162bdf92702b662f7567afc5fa7b2fdd02af pushed to master.
